homeassistant/components/voip/devices.py

```diff
@@ -136,7 +136,11 @@ def async_get_or_create(self, call_info: CallInfo) -> VoIPDevice:
             fw_version = None
 
         dev_reg = dr.async_get(self.hass)
-        voip_id = call_info.caller_ip
+        voip_id = (
+            call_info.caller_endpoint.uri
+            if call_info.caller_endpoint
+            else call_info.caller_ip
```

Comment:
Can we do an auto-migration here? So every incoming device, we check if it exists under the old ID, and if so, we migrate it to the new endpoint uri.

Reply:
I tried to update it to do an auto-migration, but it would be difficult for me to test that since I already have new devices for all of my phones.